prime-dotnet-webapi/Models/Api/PaginatedList.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+
+namespace Prime.Models.Api
+{
+    public class PaginatedResponse<T>
+    {
+        public List<T> Results { get; set; }
+        public int Page { get; set; }
+        public int TotalPages { get; set; }
+        public int TotalResults { get; set; }
+        public int PageSize { get; set; }
+
+        public PaginatedResponse(PaginatedList<T> list)
+        {
+            Results = list;
+            Page = list.Page;
+            TotalPages = list.TotalPages;
+            TotalResults = list.TotalResults;
+            PageSize = list.PageSize;
+        }
+    }
+    public class PaginatedList<T> : List<T>
+    {
+        private const int _pageSize = 100;
+
+        public int Page { get; private set; }
+        public int TotalPages { get; private set; }
+        public int TotalResults { get; private set; }
+        public int PageSize { get; private set; }
+
+        public PaginatedList(IEnumerable<T> items, int count, int page, int pageSize)
```

Comment:
I am wondering if this constructor should be `private`?

Reply:
Not sure if it should or shouldnt, but can switch it private since currently it is only being called from the static method.